server/start.go

```diff
@@ -115,6 +115,11 @@ which accepts a path for the resulting pprof file.
 
 			// amino is needed here for backwards compatibility of REST routes
 			err := startInProcess(serverCtx, clientCtx, appCreator)
+			if _, ok := err.(ErrorCode); ok {
+				serverCtx.Logger.Info(fmt.Sprintf("received quit signal: %s", err))
```

Comment:
if `ErrorCode.Code != 0` this would unconditionally exit gracefully - is that really what we want?

Reply:
We should exit with the status code from the signal.

Reply:
the problem is that other errors that are not of type ErrorCode can come from startInProcess, only the ErrorCode comes when it is killed by Signal.

Before it returned as error so cobra by default showed the help message.

```
// WaitForQuitSignals waits for SIGINT and SIGTERM and returns.
func WaitForQuitSignals() ErrorCode {
	sigs := make(chan os.Signal, 1)
	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
	sig := <-sigs
	return ErrorCode{Code: int(sig.(syscall.Signal)) + 128}
}
```

Reply:
That's fine, just exist with status code `1` when not a `ErrorCode` error.

Reply:
> just exist with status code 1 when not a ErrorCode error.

That sounds like the right solution @jgimeno. When we handle the signal, we should exit with 0. So when `!ok` just `return err`.